__test__/integration/debt_token_api/scenarios/get_approved.ts

```diff
@@ -1,17 +1,63 @@
 import { DebtTokenScenario } from "./scenarios";
 import { Orders } from "./orders";
+import { DebtTokenAPIErrors } from "src/apis/debt_token_api";
 
-export const GET_APPROVED_SCENARIOS: DebtTokenScenario.GetApprovedScenario[] = [
-    {
-        description: "debt token specifies an approved account",
-        orders: Orders.ALL_ORDERS,
-        ...Orders.WHO,
-        isApproved: true,
-    },
-    {
-        description: "debt token does not specify an approved account",
-        orders: Orders.ALL_ORDERS,
-        ...Orders.WHO,
-        isApproved: false,
-    },
-];
+const defaults = {
+    orderFilledByCreditorOne: Orders.CREDITOR_ONE_ORDER,
+    orderFilledByCreditorTwo: Orders.CREDITOR_TWO_ORDER,
+    tokenID: (creditorOneTokenID, creditorTwoTokenID, nonexistentTokenID, malFormedTokenID) =>
+        creditorOneTokenID,
+    approvee: Orders.APPROVEE,
+};
+
+const successfulDefaults = {
+    ...defaults,
+    shouldSucceed: true,
+};
+
+const unsuccessfulDefaults = {
+    ...defaults,
+    shouldSucceed: false,
+    isApproved: false,
+};
+
+export namespace GetApprovedScenarios {
+    export const SUCCESSFUL: DebtTokenScenario.GetApprovedScenario[] = [
```

Comment:
I like this pattern of putting the successful / unsuccessful scenarios in a common namespace 👌

Reply:
💯